examples/core_api/timm-post/post.md

```diff
@@ -0,0 +1,286 @@
+(This post references unreleased features, namely https://github.com/determined-ai/determined/pull/3859 and https://github.com/determined-ai/determined/pull/3807.  To run examples, you'll need to create a local branch with those two PRs merged.)
+
+(TODO: Replace github references with doc references where possible.)
+
+(TODO: Are the initial paragraphs the right story to tell about Core API?  Could completely replace them.)
+
+Users of Determined have always faced a small but annoying barrier to entry - porting their model training scripts to use one of Determined's Trial APIs.  Similar to popular libraries like PyTorch Lightning, our Trial APIs provide user-definable hooks for each step in a standard model training workflow.  Our behind-the-scenes execution harness then handles the messy details of providing scalable, distributed, preemptible training with full access to features like hyperparameter search and checkpoint management.
+
+Over time, we've realized that ML teams often have extensive script libraries and custom execution harnesses of their own that make this porting process non-trivial.  To address this, we designed *Core API*: an API with fewer built-in assumptions that can be plugged into existing code with minimal fuss.  We're releasing it as part of Determined 0.17.XX.
```

Comment:
Let's couch this as a "huge improvement" rather than as a "big bugfix".  And that's not just marketing speak either... I don't know of any of our competitors who offer the level of platform integration that core api offers.  Things like out-of-the-box distributed communication primitives and distributed preemption via a click in the webui are, afaik, unique to us.

------

Until now, Determined has offered scheduling, distributed training, metrics reporting, checkpoint tracking, and hyperparameter search to PyTorch and Keras models.  Users only had to implement the `PyTorchTrial` or `TFKerasTrial` interface to integrate with the platform.

Still, we know many ML teams have extensive script libraries and custom training loops of their own.  Ideally, they would like to access the benefits of the Determined platform without refactoring their code as a Trial class.

So we developed the Determined Core API.  Released in Determined version 0.18.0, the Core API allows users to fully integrate arbitrary models and training loops into the Determined platform.  With minimal fuss, you can now imbue any training loop with all of the features offered by the Trial APIs.